app/models/surrogate_key_logging/surrogate.rb

```diff
@@ -16,8 +16,13 @@ def surrogate_for_value(value)
       end
 
       def add(surrogate, value)
-        s = new(key: surrogate, value: value, hashed_value: hash_value(value))
-        s.save
+        begin
+          s = new(key: surrogate, value: value, hashed_value: hash_value(value))
+          s.save
+        rescue => e
```

Comment:
Thoughts on multiple levels of rescue? Maybe first level is for the duplicate key entry error?

And other errors would be remaining level.

For quick reference: https://blog.appsignal.com/2018/04/10/rescuing-exceptions-in-ruby.html